devito/ir/support/basic.py

```diff
@@ -270,14 +24,39 @@ class IterationInstance(LabeledVector):
     A representation of the iteration and data points accessed by an
     Indexed object. Three different concepts are distinguished:
 
-        * Index functions: the expressions describing what *iteration* space point
-          are accessed.
+        * Index functions: the expressions describing what *iteration* space
+          points are accessed.
         * ``aindices``: the Dimensions acting as iteration variables.
-          There is one aindex for each index function. If the index function
-          is non-affine, then it may not be possible to detect its aindex;
-          in such a case, None is used as placeholder.
+          There is one aindex for each non-constant affine index function. If
+          the index function is non-affine, then it may not be possible to detect
+          its aindex; in such a case, None is used as placeholder.
         * ``findices``: the Dimensions describing what *data* space point
           are accessed.
+
+    An IterationInstance may be regular or irregular. It is regular if and only
+    if *all* index functions are affine in their respective findex.  The
+    downside of irregular IterationInstances is that dependence testing is
+    harder, which in turn may require the data dependence analyzer to act more
+    conservatively.
+
+    Examples
+    --------
+    Given:
+        x, y, z : findices
+        w : a generic Dimension
+
+           | x+1 |           |  x  |          |  x  |          | w |          | x+y |
+    obj1 = | y+2 | ,  obj2 = |  4  | , obj3 = |  x  | , obj4 = | y | , obj5 = |  y  |
+           | z-3 |           | z+1 |          |  y  |          | z |          |  z  |
+
+    We have that:
+
+        * obj1 and obj2 are regular;
```

Comment:
So the 'regularity' is just determined topologically? i.e. if my given `findices` are `z, x, y` then
```     
         | a_1*z+b_1 |
obj =    | a_2*x+b_2 | 
         | a_3*y+b_3 |
```
would be regular?

Reply:
yes